### DISCUSSION TITLE
Project Template for idom

### DISCUSSION THREAD

**Discussion by seeclearvision** [idom-team/idom#651] [Question]
Is there any project template for creating a sample project to use idom?

Is there any recommendation wrt the directory structure for organising the folders within the python project?

I'm trying to get something like a boilerplate similar to what npm creates for ReactJS using node.

Cheers,

Dev.

**Comment by rmorshea** [comment]
Apps with IDOM are much simpler than React projects because they don't require any complex build tooling (e.g. using Webpack to transpile JSX or Typescript) so you're pretty much free to organize your code however you like.

With that said, we can probably make many of the same recommendation from [popular React-based projects](https://github.com/airbnb/javascript/blob/master/react/README.md). For example, limiting yourself to one component per file unless they are stateless and effectless (i.e. pure functions). However these are just conventions and certainly aren't requirements. If you have other questions about how to get started writing components I'd be happy to answer them if the [documentation](https://idom-docs.herokuapp.com/docs/creating-interfaces/index.html) isn't clear enough.

**Reply by seeclearvision** [reply]
Thanks for the reply.
Would you happen to have a simple boilerplate example which shows the usage of more than 1 component in the project?

**Reply by Archmonger** [reply]
A component can have any number of children, which are also components.

However, if you're going to have more than just parent component you may want to look the docs for [adding IDOM to an existing web server](https://idom-docs.herokuapp.com/docs/getting-started/running-idom.html#add-to-an-existing-web-server) or [embedding into an existing page](https://idom-docs.herokuapp.com/docs/getting-started/running-idom.html#embed-in-an-existing-webpage).
